langchain/document_loaders/joplin.py

```diff
@@ -0,0 +1,84 @@
+import json
+import urllib
+from datetime import datetime
+from typing import List, Optional
+
+from langchain.document_loaders.base import BaseLoader
+from langchain.schema import Document
+from langchain.utils import get_from_env
+
+
+class JoplinLoader(BaseLoader):
+    """
+    Loader that fetches notes from Joplin.
+
+    In order to use this loader, you need to have Joplin running with the
+    Web Clipper enabled (look for "Web Clipper" in the app settings).
+
+    To get the access token, you need to go to the Web Clipper options and
+    under "Advanced Options" you will find the access token.
+
+    You can find more information about the Web Clipper service here:
+    https://joplinapp.org/clipper/
+    """
+
+    def __init__(self, access_token: Optional[str] = None) -> None:
+        self.access_token = access_token or get_from_env(
+            "access_token", "JOPLIN_ACCESS_TOKEN"
+        )
+        self.get_note = (
+            f"http://localhost:41184/notes?token={self.access_token}&fields=id,parent_id,title,body,created_time,updated_time&page="
```

Comment:
Most probably, the port (it is 41184) is configurable. Maybe to add it as __inti__ param with default as 41184?

Reply:
Thanks @leo-gan , good idea. Although AFAIK it's not configurable in Joplin at the moment, perhaps if someone tries tunneling tricks they would need to adjust the host/port. Added.